### PR TITLE
Update: Make `baseConfig` to behave as other config options (fixes #3…

### LINKED ISSUE #3: Rule: Warn on late definitions of variables/functions
This rule warns when a variable or function is used before it is declared, such as:

``` js
doIt();

function doIt() {
}
```

Rule name should be `no-latedef`

### PR DESCRIPTION
**Pull request by gyandeeps**
…371)

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -59,49 +59,78 @@ function isFilePath(filePath) {
 }
 
 /**
- * Load and parse a JSON config object from a file.
+ * Check if item is an javascript object
+ * @param {*} item object to check for
+ * @returns {boolean} True if its an object
+ * @private
+ */
+function isObject(item) {
```
Comment (BYK) [inline lib/config.js]: Why not use https://www.npmjs.com/package/lodash.isplainobject instead? It does [a bit more](https://github.com/lodash/lodash/blob/master/lodash.js#L8289).

Comment (gyandeeps) [inline lib/config.js]: I am not sure if we need that. As this function is sufficient enough for our usecase. It also saves an additional dependency.
Am I missing any usecase?

Comment (BYK) [inline lib/config.js]: I agree. I just wanted to point out if you preferred that. The trade off for saving a dependency is us maintaining this code which looks perfectly okay for this use case :)

File: lib/config.js [hunk 2]
```diff
@@ -59,49 +59,78 @@ function isFilePath(filePath) {
...
+    return typeof item === "object" && !Array.isArray(item) && item !== null;
+}
+
+/**
+ * Read the config from the config JSON file
  * @param {string} filePath the path to the JSON config file
- * @returns {Object} the parsed config object (empty object if there was a parse error)
+ * @returns {object} config object
```
Comment (nzakas) [inline lib/config.js]: Big "O" on Object. :)

**Comment by gyandeeps** [comment]
@nzakas Should be good to go now.

**Pull request merged**